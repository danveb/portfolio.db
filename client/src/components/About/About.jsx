import profile from "../../assets/profile/profile-bw.png"; 
import { logos } from "../../constants/logos"; 
import "./About.scss"; 

const About = () => {
    return (
        <div id="about" className="about">
            <div className="about__wrapper">
                <div className="about__main">
                    <div className="about__title">
                        <h3>My Story<span>.</span></h3>
                    </div>
                    <div className="about__img">
                        <img src={profile} alt="profile pic bw" />
                    </div>
                    <div className="about__story">
                        <p>My journey started in 2020 when I wrote my first line of code, <span>Hello World</span> on an HTML document. This simple, yet powerful line of code was the catalyst to learning web development (courses on HTML, CSS & JavaScript) and thus building my first static website.</p>
                        <p>Fast-forward to today, most of my web development work is with React. I design and develop web apps that provide a rich user interface/experience. I focus on creating clean, functional, scalable, and testable apps.</p>
                        <p>On weekends I spend time tufting cool rugs, crafting home-made boba tea with friends and dreaming about soccer.</p>
                    </div>
                    <div className="about__skills">
                        <ul>
                            {logos.map((logo) => (
                                <li key={logo.id}>
                                    <img src={logo.img} alt={logo.alt} />
                                </li>
                            ))}
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default About