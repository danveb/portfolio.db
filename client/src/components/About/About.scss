.about {
    display: flex; 
    justify-content: center; 
    align-items: center; 
    flex-direction: column; 
    background-color: var(--raisin-black);
    padding: 40px 0; 

    @media screen and (max-width: 768px) {
        padding: 30px 20px; 
    }

    @media screen and (max-width: 480px) {
        padding: 25px 5px; 
    }

    .about-wrapper {
        flex: 1;
        display: flex; 
        max-width: 1200px; 
        margin: 40px; 
        padding: 30px; 
    
        @media screen and (max-width: 768px) {
            flex-direction: column; 
            margin: 20px; 
        }

        @media screen and (max-width: 480px) {
            margin: 15px; 
            padding: 25px 10px; 
        }
    
        .about-left {
            flex: 1; 
            display: flex; 
            justify-content: center; 
            align-items: center; 
            flex-direction: column; 
    
            img {
                height: auto; 
                width: 100%; 
                object-fit: cover; 
                border-radius: 10px; 
    
                @media screen and (max-width: 768px) {
                    width: auto; 
                    height: 300px; 
                }
    
                @media screen and (max-width: 480px) {
                    height: 250px; 
                }
            }
        }
    
        .about-right {
            color: var(--alice-blue); 
            flex: 1; 
            display: flex; 
            justify-content: center; 
            flex-direction: column;    
            margin: 40px; 
    
            @media screen and  (max-width: 768px) {
                margin: 20px 0px; 
            }
    
            .about-title {
                h3 {
                    font-size: 26px; 
                    font-weight: bold; 

                    span {
                        color: var(--bottle-green); 
                    }

                    @media screen and (max-width: 786px) {
                        font-size: 22px; 
                    }
                }
            }
    
            .about-description {
                p {
                    line-height: 30px; 
                    margin: 20px 0;
    
                    @media screen and (max-width: 768px) {
                        font-size: 14px; 
                    }
                }
            }

            .about-skills {
                ul {
                    display: flex; 
                    align-items: center; 
                    justify-content: center; 
                    flex-wrap: wrap; 
                    gap: 30px; 

                    img {
                        height: 40px; 
                        width: 40px; 
                        object-fit: fit; 
                    }

                    @media screen and (max-width: 768px) {
                        gap: 25px; 
                    }
                }
            }
        }
    }
}