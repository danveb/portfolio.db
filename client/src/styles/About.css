.about {
  width: 100%; 
  height: 100%; 
}

.about__wrapper {
  max-width: 768px; 
  margin: 0 auto; 
  width: 100%; 
  height: 100%; 
  background-color: var(--rich-black); 
}

.about__main,
.about__skills {
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  gap: 20px; 
  margin: 40px 0;
}

.about__title {
  font-family: "Mulish", sans-serif; 
  font-size: 20px; 
  color: var(--alice-blue); 
  letter-spacing: 0.5px; 
}

.about__profile {
  max-height: 500px; 
  display: flex; 
  align-items: center; 
  justify-content: center; 
}

.about__profile img {
  width: 50%; 
  height: 50%; 
  object-fit: contain; 
}

.about__intro {
  font-family: inherit; 
  font-size: 16px; 
  font-weight: 300; 
  color: #d4d4d4; 
  letter-spacing: 1px; 
  line-height: 1.5rem; 
}

.about__intro p span {
  color: #fff;
  font-weight: 600; 
}

.about__skills ul {
  display: flex; 
  align-items: center; 
  justify-content: center; 
  flex-wrap: wrap; 
  gap: 30px; 
}

.about__skills li {
  list-style: none; 
}

.about__skills img {
  height: 45px; 
  width: 45px; 
  object-fit: contain; 
}

@media screen and (max-width: 1024px) {
  .about__wrapper {
    padding: 0 20px; 
  }
}

@media screen and (max-width: 768px) {
  .about__title {
    font-size: 16px; 
  }

  .about__intro {
    font-size: 14px; 
  }
}