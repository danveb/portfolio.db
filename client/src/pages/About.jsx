import Button from "../components/Button";
import profile from "../assets/profile/profile-bw.png"; 
import { logos } from "../constants/logos"; 
import "../styles/About.css"; 

export default function About() {
  return (
    <div className="about">
      <div className="about__wrapper">
        <section className="about__main">
          <div className="about__title">
            <h1>this is my story<span className="dot--green">.</span> ðŸ““</h1>
          </div>
          <div className="breakline"></div>
          <div className="about__profile">
            <img src={profile} alt="b&w profile avatar" />
          </div>
          <div className="about__intro">
            <p>I started my journey as a developer back in 2020, when I wrote my first line of code, <span>Hello World</span> on an HTML document. Little did I know this simple, yet powerful line of code would motivate me to learn web development due to its limitless posibilities in the technology space. I delved deeper into HTML, CSS and JavaScript, laying the foundation for my journey in web development as a developer. </p>
          </div>
          <div className="about__intro">
            <p>Fast-forward to today, my projects showcase a diverse range of web applications, mostly developed with React. My mission is to design and develop web applications that provide a rich user interface. I focus on creating an aesthetically clean, functional, scalable, and testable apps. </p>
          </div>
          <div className="about__intro">
            <p>I aspire to continue my journey learning TypeScript, Next.js and more, to be able to develop applications that solve problems. </p>
          </div>
        </section>
        <section className="about__skills">
          <ul>
            {logos.map((logo) => (
              <li key={logo.id}>
                <img src={logo.img} alt={logo.alt} />
              </li>
            ))}
          </ul>
        </section>
        <section className="about__bottom">
          <Button title="explore projects" url="/projects" />
        </section>
      </div>
    </div>
  )
}