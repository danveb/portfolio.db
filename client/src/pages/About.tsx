import Button from "../components/Button";
import profile from "../assets/profile/profile-bw.png"; 
import { logos } from "../constants/logos"; 
import "../styles/About.css"; 

export default function About() {
  return (
    <div className="about">
      <div className="about__wrapper">
        <section className="about__main">
          <div className="about__title">
            <h1>this is my story<span className="dot--green">.</span> ðŸ““</h1>
          </div>
          <div className="breakline"></div>
          <div className="about__profile">
            <img src={profile} alt="b&w profile avatar" />
          </div>
          <div className="about__intro">
            <p>I started my journey as a developer back in 2020, when I wrote my first line of code, <span>Hello World</span> on an HTML document. Little did I know this simple, yet powerful line of code would motivate me to learn web development due to its limitless posibilities in the technology space. I delved deeper into HTML, CSS and JavaScript, laying the foundation for my journey in web development as a developer. </p>
          </div>
          <div className="about__intro">
            <p>Fast-forward to today, my projects showcase a diverse range of web applications, mostly developed with React. I'm currently learning TypeScript to encourage and enhance type safety of my code as well as to scale existing and new projects. </p>
          </div>
          <div className="about__intro">
            <p>My mission is to design and develop web applications that provide a rich user interface. I focus on creating an aesthetically clean, functional, scalable, and testable apps. </p>
          </div>
        </section>
        <section className="about__skills">
          <ul>
            {logos.map((logo) => (
              <li key={logo.id}>
                <img src={logo.img} alt={logo.alt} />
              </li>
            ))}
          </ul>
        </section>
        <section className="about__bottom">
          <Button title="explore projects" url="/projects" />
        </section>
      </div>
    </div>
  )
}