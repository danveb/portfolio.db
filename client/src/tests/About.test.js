import { render } from "@testing-library/react"; 
import About from "../components/About/About"; 

// smoke test
test("renders without crashing", () => {
    render(<About />); 
}); 

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(<About />); 
    expect(asFragment()).toMatchSnapshot(); 
});

// getByText 
// <h3>My Story<span>.</span></h3> 
test("About component getByText", () => {
    const { getByText } = render(<About />); 
    getByText("My Story"); 
}); 

// getByAltText
// <img src={logo.img} alt="html logo" />
test("About component getByAltText", () => {
    const { getByAltText } = render(<About />); 
    getByAltText("html logo"); 
})