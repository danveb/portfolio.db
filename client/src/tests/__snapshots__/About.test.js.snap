// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="about"
    id="about"
  >
    <div
      class="about-wrapper"
    >
      <div
        class="about-left"
      >
        <img
          alt="barista"
          src="profile.jpg"
        />
      </div>
      <div
        class="about-right"
      >
        <div
          class="about-title"
        >
          <h3>
            My Story
            <span>
              .
            </span>
          </h3>
        </div>
        <div
          class="about-description"
        >
          <p>
            My journey started in 2020 when I wrote my first line of code, 
            <span>
              Hello World
            </span>
             on an HTML document. This simple, yet powerful line of code was the catalyst to learning web development (courses on HTML, CSS & JavaScript) and thus building my first static website.
          </p>
          <p>
            Fast-forward to today, most of my web development work is with React. I design and develop web apps that provide a rich user interface/experience. I focus on creating clean, functional, scalable, and testable apps.
          </p>
          <p>
            On weekends I spend time tufting cool rugs, crafting home-made boba tea with friends and dreaming about soccer.
          </p>
        </div>
        <div
          class="about-skills"
        >
          <ul>
            <li>
              <img
                alt="html logo"
                src="html5.png"
              />
            </li>
            <li>
              <img
                alt="css logo"
                src="css.png"
              />
            </li>
            <li>
              <img
                alt="javascript logo"
                src="js.png"
              />
            </li>
            <li>
              <img
                alt="react logo"
                src="react.png"
              />
            </li>
            <li>
              <img
                alt="sass logo"
                src="sass.png"
              />
            </li>
            <li>
              <img
                alt="express logo"
                src="express.png"
              />
            </li>
            <li>
              <img
                alt="psql logo"
                src="psql.png"
              />
            </li>
            <li>
              <img
                alt="mongoDB logo"
                src="mongoDB.png"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
